python/tskit/trees.py

```diff
@@ -2903,6 +2903,52 @@ def kc_distance(self, other, lambda_=0.0):
         """
         return self._ll_tree.get_kc_distance(other._ll_tree, lambda_)
 
+    def _get_sample_sets(self):
+        ret = collections.defaultdict(set)
+        for u in self.samples():
```

Comment:
We should try to avoid making a separate copy of these as frozen sets. How about:
```
def _get_sample_sets(self):
     ret = {}
     for u in self.nodes(order="postorder"):
           u_sample_set = set()
           if self.is_sample(u):
                u_sample_set.add(u)
           for  v in self.children(u):
                 u_sample_set |= ret[v]
            ret[u] = frozenset(u_sample_set)
```
Then we should be able to work directly with the return value of this function?